src/utils/EventUtils.js

```diff
@@ -50,3 +51,35 @@ export function canEditContent(mxEvent) {
         mxEvent.getOriginalContent().msgtype === "m.text" &&
         mxEvent.getSender() === MatrixClientPeg.get().getUserId();
 }
+
+export function findPreviousEditableEvent(room, fromEventId = undefined) {
+    const liveTimeline = room.getLiveTimeline();
+    const events = liveTimeline.getEvents();
+    let startFromIdx = events.length - 1;
+    if (fromEventId) {
+        const fromEventIdx = findLastIndex(events, e => e.getId() === fromEventId);
+        if (fromEventIdx !== -1) {
+            startFromIdx = fromEventIdx - 1;
+        }
+    }
+    const nextEventIdx = findLastIndex(events, e => !shouldHideEvent(e) && canEditContent(e), startFromIdx);
```

Comment:
imo `canEditContent` should only check if the `sender === myUserId` because it'll be really awkward to press the up arrow and edit someone else's message (because you happen to be a moderator when we support this)

Reply:
I'm not sure I understand what you suggest to change. `canEditContent` *does* check the sender is the current user. It also filters events to be `type=m.room.message` and `msgtype=m.text`, because we only provide the edit UI for those messages, so not filtering for this would break the UI?

We also need `shouldHideEvent` here because otherwise edits (m.replace relations), which are not rendered but of same type and msgtype, would be attempted to edit.

Reply:
canEditContent does the check now, but in the future it probably wouldn't. When we support moderators editing messages, changing that function would be a good starting point, which would cause a bug in the up/down arrow logic. It's just to prevent a future bug. 

I guess if it filters the events down further then a third condition on this line would be nice for the sender.

Reply:
Oh, I see what you mean. Added a `canEditOwnEvent` that just calls `canEditContent` for now with a note. Good enough for now?

Reply:
Yup - thanks!